### DISCUSSION TITLE
Are presets intended to be used alongside plugins?

### DISCUSSION THREAD

**Discussion by salieflewis** [dotansimha/graphql-code-generator#9650] [Q&A]
I'm confused about whether or not the `graphql-request` client preset is providing more functionality in addition to what I've already unlocked by using the `typescript`, `typescript-operations`, and `typescriot-graphql-request` plugins.

There seems to be a bit overlap and I'm having some issues using the cli with those things together.

**Comment by charpeni** [comment]
`client-preset` (which is newer) depends on `typescript` and `typescript-operations` plugins and could be seen as a simple opinionated configuration and a lightweight types-only generation.

> ### A unified configuration and package for all GraphQL clients
> 
> Most of the existing client-side plugins (`typescript-react-apollo`, `typescript-react-query`, etc) rely on the generation of hooks or SDKs that wrap the underlying GraphQL Client in a type-safe way.
> 
> However, the generation of hooks or SDK code brings many downsides:
> 
> - an unnecessary increase of the final bundle size
> - misalignment between the generated hooks signature and the underlying GraphQL Client
> - inconsistencies of configuration options and preset compatibility across packages (ex: `near-operation-file compatibility`)
> 
> To make GraphQL code generation great and simple, the v3 version will introduce two major changes:
> 
> - **a new unique preset for all GraphQL clients, which include better developer experience, smaller bundle size, stronger typings, and easier-to-follow best practices**
> - **a TypeScript-first configuration file** that will allow configuration autocompletion
>
> Source: https://github.com/dotansimha/graphql-code-generator/issues/9456.

**Reply by salieflewis** [reply]
Are you aware if `client-preset` suffers from compatibility issues alongside the `graphql-request` plugin?

**Reply by finkpong** [reply]
i spent some time just trying to understand what benefits client-preset provided over the old impl (typescript-operations) coz at first glance there was something lost when implementing the client-preset way, i.e. writing graphql code directly in react tsx code, vs normal function calls with typescript-operations. 
i wish the justifications written in https://github.com/dotansimha/graphql-code-generator/issues/9456 was written into the official docs, could have saved me a lot of time and frustration
